support/cas-server-support-couchdb-core/src/main/java/org/apereo/cas/couchdb/core/CouchDbConnectorFactory.java

```diff
@@ -23,19 +26,69 @@
 @Getter
 @Setter
 @RequiredArgsConstructor
+@Slf4j
 public class CouchDbConnectorFactory {
 
-    private final AbstractCouchDbProperties couchDbProperties;
+    private final BaseCouchDbProperties couchDbProperties;
+    private final ObjectMapperFactory objectMapperFactory;
     private CouchDbConnector couchDbConnector;
     private CouchDbInstance couchDbInstance;
+    private HttpClient httpClient;
```

Comment:
It really looks like these objects are treated as singletons? If so, why not make them be singletons by following that pattern?

Reply:
Not my classes... They wouldn't be singletons if a user put different databases in different couchdb servers. I might wind up doing that to separate the higher traffic ones from the lower...

Reply:
lazy initialization, it's so the create methods can be called more than once... probably should rename them...

Reply:
Done